jre_emul/Classes/com/google/j2objc/security/IosRSAKeyPairGenerator.java

```diff
@@ -74,10 +85,31 @@ public native KeyPair generateKeyPair() /*-[
     JavaSecurityKeyPair *keyPair =
         AUTORELEASE([[JavaSecurityKeyPair alloc] initWithJavaSecurityPublicKey:publicKey
                                                     withJavaSecurityPrivateKey:privateKey]);
+
     [publicKey release];
     [privateKey release];
     return keyPair;
   ]-*/;
+  
+  /*-[
+  -(BOOL) deleteKey:(NSString *)tag   
+  		   keyClass:(CFStringRef) keyClass {
+  		   
+    NSData *publicTag = [tag dataUsingEncoding:NSUTF8StringEncoding];
+
+    NSMutableDictionary *query = [NSMutableDictionary dictionary];
+    query[(id)kSecClass] = (id)kSecClassKey;
+    query[(id)kSecAttrKeyType] = (id)kSecAttrKeyTypeRSA;
+    query[(id)kSecAttrKeyClass] = (id)keyClass;
+    query[(id)kSecAttrApplicationTag] = tag;
+	OSStatus status2 = SecItemDelete((CFDictionaryRef) query);
+    if (status2 != errSecSuccess && status2 != errSecItemNotFound) {
+        NSLog (@"Problem removing previous public key from the keychain, OSStatus == %d", (int)status2);
+        return false;
+    }
```

Comment:
Ah, sorry, the import is missing because there's no reference to java.security.ProviderException in the Java code in this class (notice how it is referenced in IosRSAKey.java, so this function is declared there).  Add that import as follows:
```
@@ -26,6 +26,7 @@ import java.security.spec.RSAKeyGenParameterSpec;
 
 /*-[
 #import "com/google/j2objc/security/IosRSAKey.h"
+#import "java/security/ProviderException.h"
 ]-*/
 
 public class IosRSAKeyPairGenerator extends KeyPairGeneratorSpi {
```